source/common/upstream/cds_subscription.h

```diff
@@ -55,6 +56,7 @@ class CdsSubscription : public Http::RestApiFetcher,
   Config::SubscriptionCallbacks<envoy::api::v2::Cluster>* callbacks_ = nullptr;
   Config::SubscriptionStats stats_;
   const absl::optional<envoy::api::v2::core::ConfigSource>& eds_config_;
+  const Stats::Scope& scope_;
```

Comment:
Instead of storing a reference to the scope here, would it better to store  reference to the stat options? We are slightly splitting hairs but it potentially seems a bit clearer/safer to me.

Reply:
Makes sense -- I put statsOptions into the scope struct, and found it made more contextual sense to add scope as an argument to a lot of higher-level constructors or factories. But as those factories call lower-level utilities (`parseHttpConnectionManagerFromJson`, it makes more sense to give them the specific arguments they need, like the `statsOptions` struct. I've changed the interfaces for `CdsJson::translateCluster`, `LdsJson::translateListener`, and some others.